Medbay Oxigen Assistant //WIP//

SagaStars opened the pull request:
Una maquina nueva para medbay, funciona como una IV-BAG pero funcional en base a power cells por lo cual estara curando el oxy-loss y bajando el pulso cardiaco para estabilizar al paciente.

 Tiene el proposito de ser usado como una alternativa al estar en un cryotube 10 minutos o ir comiendote todas las pastillas de salbutomol que hay. Ademas de traer posibilidades de roleplay y poder recrear el iconico **| || || |_**

**Imagenes**
No hay sprite aun.

**Changelog**
🆑 
add: Medbay Oxigen Assistant
add: El quimico que estara usando la maquina
🆑

SagaStars committed:
el papa de los commits

SagaStars committed:
Casi terminado

- Ahora gasta energia de la cell y puede descargarse.
- Las EMP le afectan y drenan toda la energia.
- Hace un sonido fastidioso cuando esta conectado a un paciente sin energia.
- Retoques al quimico
- Un drip para el moa (W.I.P)

Evankhell561 commented on code/HISPANIA/modules/reagents/reagent_containers/moa.dm:
```diff
@@ -0,0 +1,106 @@
+	w_class = WEIGHT_CLASS_HUGE
+	var/status = 3
+	var/hitcost = 25
+	var/obj/item/stock_parts/cell/high/bcell = null
```
Cuando hablamos en discord, no me di cuneta de esto. 
ponerle var/obj/item/stock_parts/cell/high/bcell = null puede darte problemas cuando quieras cambiarle la bateria y con algunas cosillas más. 
cambialo por 
var/obj/item/stock_parts/cell/bcell = null

SagaStars replied:
Understandable

Evankhell561 commented:
Tengo, algunas preguntas, ¿Donde consigues este aparato? ¿donde consigues el medicamento?
recomendaria que la maquina solo pueda pedirse por cargo/ciencias y que no esté desde el inicio. 
¿se podrán producir los quimicos que estás usando?
no me gustaria que un quimico pueda hacer esto, a menos que el quimico solo funcione en esta maquina y no funcione normalmente (o si funciona que funcione en el cryotubo)

Evankhell561 commented on code/HISPANIA/game/machinery/moa_drip.dm:
```diff
@@ -0,0 +1,41 @@
+			var/image/filling = image('icons/hispania/obj/moa_drip.dmi', src, "hangingbag-fluid")
+			add_overlay(filling)
+
+/obj/machinery/iv_drip/attackby(obj/item/I, mob/user, params)
```
esto deberia ser /obj/machinery/iv_drip/moa/attackby(obj/item/I, mob/user, params)
en vez de /obj/machinery/iv_drip/attackby(obj/item/I, mob/user, params). Al hacerlo como lo estás haciendo tú, afectas al iv_drip original.

SagaStars replied:
Si, justo me fije de esto luego hacer el commit. Pronto lo arreglo.

Evankhell561 commented on code/HISPANIA/modules/reagents/reagent_containers/moa.dm:
```diff
@@ -0,0 +1,106 @@
+			bcell.use(hitcost)
+			update_icon()
+
+/obj/item/reagent_containers/iv_bag/attack_self(mob/user)
```
/obj/item/reagent_containers/iv_bag/moa/attack_self(mob/user)

SagaStars replied:
Si, justo me fije de esto luego hacer el commit. Pronto lo arreglo.

Evankhell561 commented on code/HISPANIA/modules/reagents/reagent_containers/moa.dm:
```diff
@@ -0,0 +1,106 @@
+	volume = 10000
+	w_class = WEIGHT_CLASS_HUGE
+	var/status = 3
+	var/hitcost = 25
```
Con un hit cost de 25 kW y una bateria de 10kW esto tardará unos 400 ticks en gastarse, lo que se traduce a 800 segundos, lo que son unos 13 minutos y eso representa todo el tiempo que vayas a usar esta cosa y mucho más. Subelo un poco da la oportunidad a que se le acabe la bateria y que conseguir una mejor sea importante. Yo lo subiria a var/hitcost = 110 para que dure como 3 minutos con una beteria como la que le estás colocando inicialmente. o ponerle var/hitcost = 66 para que dure como 5 minutos (aunque me parece mucho pero eso ya es decision tuya).

SagaStars replied:
Pues si reconsiderandolo es demasiado tiempo para una version con un cell basica, lo pondre en 64.

SagaStars committed:
Review gamer

- Gasto  de energia de la cell ajustada a 64.
- Nueva ruta para la variable var/obj/item/stock_parts/cell/high.

SagaStars commented:
> Tengo, algunas preguntas, ¿Donde consigues este aparato? ¿donde consigues el medicamento?
> recomendaria que la maquina solo pueda pedirse por cargo/ciencias y que no esté desde el inicio.
> ¿se podrán producir los quimicos que estás usando?
> no me gustaria que un quimico pueda hacer esto, a menos que el quimico solo funcione en esta maquina y no funcione normalmente (o si funciona que funcione en el cryotubo)

- Solo vendria una maquina en medbay por defecto y el medicamento no podria ser creado.
- Tengo la idea de que pueda ser pedida por cargo a un precio relativamente alto (como el de un hardsuit) y sobre ciencias, seria a niveles altos. 
- No.
- Misma idea tengo, por ahora estoy viendo soluciones a este problema como que necesites tal temperatura y la maquina te suba y tal.

Evankhell561 commented on code/HISPANIA/modules/reagents/reagent_containers/moa.dm:
```diff
@@ -7,8 +7,8 @@
-	var/hitcost = 25
-	var/obj/item/stock_parts/cell/high/bcell = null
+	var/hitcost = 64
+	var/obj/item/stock_parts/cell/high = null
```
travis te tiró error porque lo que yo dije fue que cambiaras 
var/obj/item/stock_parts/cell/high/bcell  por var/obj/item/stock_parts/cell/bcell. 
En cambio tu lo que hiciste fue cambiarlo por var/obj/item/stock_parts/cell/high.

SagaStars replied:
![IMG_0275](https://user-images.githubusercontent.com/55407528/67247584-dd014900-f42f-11e9-8352-4499a7d8b454.jpeg)

SagaStars committed:
Bro momento

- Arregla la ruta de la variable de antes.

Evankhell561 commented:
Pues te diria que mejor la metieras en cargo, hacerla en ciencias genera mucho problema. Hacer un pedido es más sencillo.

SagaStars commented:
Lo tendre en consideracion *wink*

Evankhell561 commented on code/HISPANIA/modules/reagents/reagent_containers/moa.dm:
```diff
@@ -0,0 +1,106 @@
+	w_class = WEIGHT_CLASS_HUGE
+	var/status = 3
+	var/hitcost = 64
+	var/obj/item/stock_parts/cell/bcell = null
```
no es obligatorio pero puedes quitar todo lo que diga "bcell", y reemplazarlo por "cell"? Funcionará igual y no cambiará nada pero odio el "bcell". Acá un ejemplo: 
var/obj/item/stock_parts/cell/bcell = null
var/obj/item/stock_parts/cell/cell= null

SagaStars committed:
Update moa.dm

SagaStars committed:
Merge branch 'oxygenador_de_sangre' of https://github.com/SagaStars/Paradise into oxygenador_de_sangre

SagaStars committed:
bruh

- quite la f

SagaStars commented:
acabo de hacer un whoopsie.
